compiler/main/driver.cpp

```diff
@@ -1936,6 +1942,13 @@ static void warnDeprecatedFlags() {
 
 // Check for inconsistencies in compiler-driver control flags
 static void checkCompilerDriverFlags() {
+  // Warn in initial driver invocation if extraneous --compiler-driver was set
+  if (driverModeSpecified && !fDriverDoMonolithic && !driverInSubInvocation) {
+    USR_WARN(
+        "--compiler-driver is deprecated; driver mode is now on by default, "
+        "use --no-compiler-driver to disable it");
```

Comment:
I'm not so sure that this warning is necessary at all. The reason is, independent of the default, IMO it's reasonable to generally have `--no-bla` and `--bla` flags. Anyway I have no problem adding this warning if you see benefit to it; I'm just saying the benefit of it isn't obvious to me.

That said, I can see benefit to an unstable warning (or something) for both `--no-compiler-driver` and `--compiler-driver`. I would expect that in the long term, we will entirely remove the monolithic compilation mode.